### Теория вероятностей и математическая статистика (семинары).  
### Урок 7. Непараметрические тесты.  
1-4 задачи решать с помощью функций, 5ю вручную.
#### Выберете тест и проверьте, есть ли различия между выборками:
#### Задача 1.  
Даны две независимые выборки. Не соблюдается условие нормальности.  
x1 380,420, 290  
y1 140,360,200,900  
Сделайте вывод по результатам, полученным с помощью функции.  
#### Решение:  


In [20]:
import math
import numpy as np
import scipy.stats as stats

x1 = [380, 420, 290]
y1 = [140, 360, 200, 900]
a=0.05
result = stats.mannwhitneyu(x1, y1)
print(f'{result}')

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)


#### Ответ:   
Pvalue=0,6286 > a=0.05, нулевая гипотеза не отвергается на уровне значимости a=0.05.

#### Задача 2.  
Исследовалось влияние препарата на уровень давления пациентов.  
Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут.  
Есть ли статистически значимые различия?  
1е измерение до приема препарата: 150, 160, 165, 145, 155  
2е измерение через 10 минут: 140, 155, 150, 130, 135  
3е измерение через 30 минут: 130, 130, 120, 130, 125  
#### Решение:

In [21]:
import math
import numpy as np
import scipy.stats as stats

before = [150, 160, 165, 145, 155]
after10 = [140, 155, 150, 130, 135]
after30 = [130, 130, 120, 130, 125]
a = 0.05

result = stats.friedmanchisquare(before, after10, after30)

print(f'{result}')

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)


#### Ответ:  
Pvalue=0.0083 < a=0.05, статистически значимые различия обнаружены на уровне значимости a=0.05.

#### Задача 3.  
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.  
1е измерение до приема препарата: 150, 160, 165, 145, 155  
2е измерение через 10 минут: 140, 155, 150, 130, 135  
#### Решение:

In [22]:
import math
import numpy as np
import scipy.stats as stats

before = [150, 160, 165, 145, 155]
after10 = [140, 155, 150, 130, 135]
a = 0.05

result = stats.wilcoxon(before, after10)

print(f'{result}')

WilcoxonResult(statistic=0.0, pvalue=0.0625)


#### Ответ:
Pvalue=0.0625 > a=0.05, статистически значимых различий не обнаружено на уровне значимости a=0.05.

#### Задача 4.  
Даны 3 группы учеников плавания.  
Время на дистанцию 50 м составляет:  
1 группа: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67  
2 группа: 57, 58, 69, 48, 72, 70, 68, 71, 50, 53  
3 группа: 57, 67, 49, 48, 47, 55, 66, 51, 54  
#### Решение:

In [23]:
import math
import numpy as np
import scipy.stats as stats

group1 = [56, 60, 62, 55, 71, 67, 59, 58, 64, 67]
group2 = [57, 58, 69, 48, 72, 70, 68, 71, 50, 53]
group3 = [57, 67, 49, 48, 47, 55, 66, 51, 54]
a = 0.05

result = stats.kruskal(group1, group2, group3)

print(f'{result}')

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)


#### Ответ:  
Pvalue=0.0650 > a=0.05, статистически значимых различий не обнаружено на уровне значимости a=0.05.

#### Задача 5.  
Заявляется, что партия изготавливается со средним арифметическим 2,5 см.  
Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения.  
Объем выборки 10, уровень статистической значимости 5%  
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34
#### Решение:

In [25]:
import math
import numpy as np
import scipy.stats as stats

m = 2.5
n = 10
a = 0.05
array = [2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]
mean = np.mean(array)
print(f'Средняя по выборке = {mean}')
sko=np.std(array, ddof=1)
print(f'Среднее квадратичное отклонение = {sko}')
t1 = (mean-m)/(sko/math.sqrt(n))
print(f'Наблюдаемое значение t-критерия = {t1}')
t2 = stats.t.ppf(1 - (a / 2), n - 1)
print(f'Табличное значение t-критерия = {t2}')

Средняя по выборке = 2.5279999999999996
Среднее квадратичное отклонение = 0.1572542173961923
Наблюдаемое значение t-критерия = 0.5630613661802959
Табличное значение t-критерия = 2.2621571627409915


#### Ответ:  
Нулевая гипотеза не отвергается на уровне значимости $\alpha=0.05$, так как 0.5630<2.2621.